src/main/java/com/example/animalshelter/controller/AnimalController.java

```diff
@@ -24,4 +22,9 @@ public AnimalController(AnimalService animalService) {
     public String addAnimal(@RequestBody Animal animal) {
         return animalService.add(animal) ? SUCCESSFUL_RESPONSE : FAILED_RESPONSE;
     }
+
+    @DeleteMapping("/{id}")
+    public String deleteAnimal(@PathVariable int id) throws DeleteAnimalException {
```

Comment:
I'm not sure if `id` should be `int` type. Maybe the better option would be to make it Long or Integer?

Reply:
It sounds right, I looked answer for this and mostly boxed type is used. I added null-check in the service class.